tests/helpers.py

```diff
@@ -854,6 +858,99 @@ def create_pods(pvc_objs, pod_factory, interface, pods_for_rwx=1, status=""):
     return pod_objs
 
 
+def create_build_from_docker_image(
+        image_name,
+        install_package,
+        namespace,
+        source_image='centos',
+        source_image_label='latest'
+):
+    """
+    Allows to create a build config using a Dockerfile specified as an argument
+    For eg., oc new-build -D $'FROM centos:7\nRUN yum install -y httpd',
+    creates a build with 'httpd' installed
+
+    Args:
+        image_name (str): Name of the image to be created
+        source_image (str): Source image to build docker image from,
+        Defaults to Centos as base image
+        namespace (str): project where build config should be created
+        source_image_label (str): Tag to use along with the image name,
+        Defaults to 'latest'
+        install_package (str): package to install over the base image
+
+    Returns:
+        (obj): Returns the image object
+
+    """
+    base_image = source_image + ':' + source_image_label
+    docker_file = f"FROM {base_image}\n" \
+                  f"RUN yum install -y {install_package}\n" \
+                  f"CMD tail -f /dev/null"
+    command = f"new-build -D $\'{docker_file}\' --name={image_name}"
+    kubeconfig = os.getenv('KUBECONFIG')
+
+    oc_cmd = f"oc -n {namespace} "
+
+    if kubeconfig:
+        oc_cmd += f"--kubeconfig {kubeconfig} "
+    oc_cmd += command
+    logger.info(f'Running command {oc_cmd}')
+    result = run(
```

Comment:
Usage of subprocess.run here is intentional as the existing function `run_cmd` uses `shlex.split` before sending the command, which removes `"` in the command being sent.

Reply:
there are other ways to fix this bug +1 on this usage